astropy/visualization/wcsaxes/patches.py

```diff
@@ -19,18 +20,19 @@
 
 # Monkey-patch the docs to fix CapStyle and JoinStyle subs.
 # TODO! delete when upstream fix matplotlib/matplotlib#19839
-Polygon.__init__.__doc__ = Polygon.__init__.__doc__.replace(
-    "`.CapStyle`", "``matplotlib._enums.CapStyle``"
-)
-Polygon.__init__.__doc__ = Polygon.__init__.__doc__.replace(
-    "`.JoinStyle`", "``matplotlib._enums.JoinStyle``"
-)
-Polygon.set_capstyle.__doc__ = Polygon.set_capstyle.__doc__.replace(
-    "`.CapStyle`", "``matplotlib._enums.CapStyle``"
-)
-Polygon.set_joinstyle.__doc__ = Polygon.set_joinstyle.__doc__.replace(
-    "`.JoinStyle`", "``matplotlib._enums.JoinStyle``"
-)
+if sys.flags.optimize < 2:
```

Comment:
What does this mean? That doc build won't be happy if `sys.flags.optimize >= 2` because this patch won't be run?

Reply:
Maybe we will never build doc in that mode anyway?

Reply:
I wouldn't recommend using it to build docs since it implies dropping docstrings completely :)